.lagoon.yml

```
@@ -1,19 +1,77 @@
 docker-compose-yaml: docker-compose.yml
 
-tasks:
-  post-rollout:
-    - run:
-        name: Deploy GovCMS
-        command: govcms-deploy
-        service: cli
-
 environments:
   master:
     types:
       mariadb: mariadb-shared
-
     cronjobs:
       - name: drush cron
         schedule: "0 * * * *"
         command: 'drush cron --root=/app'
         service: cli
+
+# Note regarding tasks in the scaffold.
+# Please avoid making scripts more than one or two lines, and avoid if/else logic.
+# When PaaS users or the support team modify this file, it should be easy to turn things on and off.
+
+tasks:
+  pre-rollout:
+    - run:
+        name: Snapshot the database and store
+        command: if [[ "$LAGOON_ENVIRONMENT_TYPE" = "production" ]]; then drush sql:dump --root=/app --gzip --result-file=/app/web/sites/default/files/private/backups/pre-deploy-dump.sql; fi
+        service: cli
+        shell: bash
+    - run:
+        name: Snapshot the config and store
+        command: if [[ $(drush cex --root=/app sync -y --destination "$TMP"/config; echo "$?") = 0 ]]; then tar -czf /app/web/sites/default/files/private/backups/pre-deploy-config.tar.gz "$TMP"/config --remove-files; fi
```

missing --quiet - otherwise will interfere with exit code
```suggestion
        command: if [[ $(drush cex --root=/app sync -y --quiet --destination "$TMP"/config; echo "$?") = 0 ]]; then tar -czf /app/web/sites/default/files/private/backups/pre-deploy-config.tar.gz "$TMP"/config --remove-files; fi
```